Marlin/src/feature/mmu/mmu2.cpp

```diff
@@ -508,9 +508,31 @@ static void mmu2_not_responding() {
 
     set_runout_valid(false);
 
-    if (index != extruder) {
+    if (index != extruder)
+    {
+      #if ENABLED(MMU_IR_UNLOAD_MOVE)
+        if (FILAMENT_PRESENT()) {
+          DEBUG_ECHOLNPGM("Unloading\n");
+          //filament_ramming(); //unloading from printer side
+          ENABLE_AXIS_E0();
+          while (FILAMENT_PRESENT())
+          {                             //if filament is still present, keep unloading
+            current_position.e -= 0.25; //this is a guessed value, adjust to your preference
+            line_to_current_position(MMM_TO_MMS(120));
+            planner.synchronize();
+          }
+          DISABLE_AXIS_E0(); //disable E stepper when unloading finished
```

Comment:
Does this have to happen before `manage_response`? If not, you can restor the old `DISABLE_AXIS_E0` and delete the three new copies.

Reply:
Oh shoot, I didn't realize that when I was coding 😅 thanks!